File: leetcode/03100_water_bottles_ii.rs
impl Solution {
    pub fn max_bottles_drunk(num_bottles: i32, num_exchange: i32) -> i32 {
        let mut drunk = 0;
        let mut cost = num_exchange;
        let mut bottles_left = num_bottles;
        while bottles_left >= cost {
            drunk += cost;
            bottles_left -= (cost - 1);
            cost += 1;      
        }
        return drunk + bottles_left;
    }
}

File: leetcode/03100_water_bottles_ii.py
class Solution:
    def maxBottlesDrunk(self, numBottles: int, numExchange: int) -> int:
        drank = empty = numBottles
        while empty >= numExchange:
            empty -= numExchange - 1
            numExchange += 1
            drank += 1
        return drank
